packages/xrpl/src/models/transactions/AMMDeposit.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable complexity -- required for validateAMMDeposit */
+import { ValidationError } from '../../errors'
+import { Amount, IssuedCurrencyAmount } from '../common'
+
+import {
+  BaseTransaction,
+  isAmount,
+  isIssuedCurrency,
+  validateBaseTransaction,
+} from './common'
+
+/**
+ * AMMDeposit is the deposit transaction used to add liquidity to the AMM instance pool,
+ * thus obtaining some share of the instance's pools in the form of LPToken.
+ *
+ * The following are the recommended valid combinations:
+ * - LPToken
+ * - Asset1In
+ * - Asset1In and Asset2In
+ * - Asset1In and LPToken
+ * - Asset1In and EPrice
+ */
```

Comment:
I would recommend including what each of these specific combinations will do in this docstring (As specified in section "2.3.1.4. Specifying different parameters" of the spec)

Ex. - LPToken - Just specifying LPToken will cause a proportional deposit of pools assets in exchange for the specified amount of LPTokens of the AMM instance.